water_mark/water_mark.py

```diff
@@ -0,0 +1,284 @@
+                if ('.dbf') in f:
+                    os.remove(os.path.join(inputsPath, f))
+
+
+def move_sar_mask_to_outDir(outDir) -> None:
+    for sar in os.listdir(outDir):
+        inputsPath = os.path.join(os.getcwd(), 'inputs')
+        image = os.path.join(inputsPath, sar + '.tif')
+        mask = os.path.join(inputsPath, sar + '_Mask.tif')
+        program = os.path.join(inputsPath, 'gdal_reclassify.py')
+        copyLocation = os.path.join(outDir, sar)
+        shutil.copy(image, copyLocation)
+        os.rename(os.path.join(outDir, sar, sar+'.tif'),
+                               os.path.join(outDir, sar, sar+'_Image.tif'))
+        shutil.copy(mask, copyLocation)
+        shutil.copy(program, copyLocation)
+        os.remove(mask)
+
+
+def cut_sar_to_tiles(outDir, mxmTileSize) -> None:
+    for sar in os.listdir(outDir):
+        image = os.path.join(outDir, sar,  sar + '_Image.tif')
+        imageData = gdal.Open(image)
+        xStep, yStep = mxmTileSize, mxmTileSize
+        xSize, ySize = imageData.RasterXSize, imageData.RasterYSize
+        count = 0
+        for x in range(0, xSize, xStep):
+            for y in range(0, ySize, yStep):
+                # fileName = 'Image_<sar>_<0-9+>.tif'
+                fileName = f"Image_{sar}_'{count}.tif"
+                gdal.Translate(os.path.join(outDir, sar, fileName),
+                               image,
+                               srcWin=[x, y, xStep, yStep],
+                               format="GTiff")
+
+
+def clip_mask_to_sar(outDir) -> None:
+    for sar in os.listdir(outDir):
+        for tile in sar:
+            cutSize = tile
+            cutSizeShp = cutSize[:-4]+'.shp'  # change ext
+            input = os.path.join(os.getcwd(), sar)
+            output = 'Mask_'+tile[5:]
+            subprocess.call(f"gdaltindex {cutSizeShp} {cutSize}",
+                            shell=True)
+            subprocess.call(f"gdalwarp -cutline {cutSizeShp} -crop_to_cutline \
+                            {input} {output}",
+                            shell=True)
+            # delete junk
+            for f in os.listdir(os.path.join(outDir, sar)):
+                if ('.shp') in f:
+                    os.remove(os.path.join(outDir, sar, f))
+                if ('.shx') in f:
+                    os.remove(os.path.join(outDir, sar, f))
+                if ('.prj') in f:
+                    os.remove(os.path.join(outDir, sar, f))
+                if ('.dbf') in f:
+                    os.remove(os.path.join(outDir, sar, f))
+
+
+def trim_masks(outDir, mxmTileSize) -> None:
+    for sar in os.listdir(outDir):
+        for f in os.listdir(outDir):
+            if f == sar+'Image.tif':
+                pass
+            elif f == sar+'Mask.tif':
+                pass
+            elif f.endswith('.py'):
+                pass
+            else:
+                resizedImage = 'RS'+f
+                subprocess.call(f"gdal_translate -outsize {mxmTileSize} \
```

Comment:
You can use the python bindings instead of calling subprocess: 
https://gdal.org/python/osgeo.gdal-module.html#Translate

```python
gdal.Translate(outsize=mxmTileSize, ...
```